cmd/config/get/get.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright (c) 2019 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package get
+
+import (
+	"fmt"
+	"github.com/openshift/rosa/pkg/config"
+	"github.com/openshift/rosa/pkg/rosa"
+	"os"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	debug bool
+}
+
+var Cmd = &cobra.Command{
+	Use:   "get [flags] VARIABLE",
+	Short: "Prints the value of a config variable",
+	Long:  "Prints the value of a config variable. See 'ocm config --help' for supported config variables.",
+	Args:  cobra.ExactArgs(1),
+	Run:   run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.debug,
+		"debug",
+		false,
+		"Enable debug mode.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+
+	// Load the configuration file:
+	cfg, err := config.Load()
+	if err != nil {
+		r.Reporter.Errorf("Can't load config file: %v", err)
+		os.Exit(1)
+	}
+
+	// If the configuration file doesn't exist yet assume that all the configuration settings
+	// are empty:
+	if cfg == nil {
+		r.Reporter.Errorf("Config file doesn't exist yet")
+		os.Exit(1)
+	}
+
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	// Print the value of the requested configuration setting:
+	switch argv[0] {
+	case "access_token":
+		fmt.Fprintf(writer, "%s\n", cfg.AccessToken)
+	case "client_id":
+		fmt.Fprintf(writer, "%s\n", cfg.ClientID)
+	case "client_secret":
+		fmt.Fprintf(writer, "%s\n", cfg.ClientSecret)
+	case "insecure":
+		fmt.Fprintf(writer, "%v\n", cfg.Insecure)
+	case "refresh_token":
+		fmt.Fprintf(writer, "%s\n", cfg.RefreshToken)
+	case "scopes":
+		fmt.Fprintf(writer, "%s\n", cfg.Scopes)
+	case "token_url":
+		fmt.Fprintf(writer, "%s\n", cfg.TokenURL)
+	case "url":
+		fmt.Fprintf(writer, "%s\n", cfg.URL)
+	case "fedramp":
+		fmt.Fprintf(writer, "%v\n", cfg.FedRAMP)
+	default:
+		r.Reporter.Errorf("Unknown setting")
+		os.Exit(1)
```

Comment:
@tirthct As a suggestion, how about we move this into a `ConfigService` that has a `PrintConfig` function? That would mean we keep Cobra as a thin wrapper around our actual logic and we can significantly improve our test coverage.

As an example, assuming an interface that looks like this:

```
type ConfigService interface {
  PrintConfig(writer Writer) error
}
```

This command could be re-written as follows

```
func run(cmd *cobra.Command, argv []string) {
  r := rosa.NewRuntime()
  cfgService := config.NewConfigService()
  err := execute(cfgService)
  if err != nil {
    r.reporter.Errorf(err)
    os.Exit(1)
  }
}

func execute(cfgService config.ConfigService) {
  writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
  return cfgService.PrintConfig(writer)
}

``` 

We could then add unit tests for the `execute` function by passing in a `mock` of the `ConfigService` interface, and also for your `ConfigService` implementation.

WDYT?

Reply:
@robpblake I am guessing the purpose here was to have as much code coverage as possible, and we have added tests to make sure we can check the entire config save/print functionality, instead of mocking these methods. Hence, we have not added the interface as of now.
@tylercreller